include/libopencm3/stm32/h7/rcc.h

```diff
@@ -0,0 +1,458 @@
+#define RCC_CFGR_MCO2_HSE         0x2
+#define RCC_CFGR_MCO2_PLL1        0x3
+#define RCC_CFGR_MCO2_CSI         0x4
+#define RCC_CFGR_MCO2_LSI         0x5
+
+/* MCO1/2PRE: MCO Prescalers */
+#define RCC_CFGR_MCOPRE_MASK      0xf
+#define RCC_CFGR_MCO2PRE_SHIFT    25
+#define RCC_CFGR_MCO1PRE_SHIFT    18
+#define RCC_CFGR_MCOPRE_DIV_NONE  0x0
+#define RCC_CFGR_MCOPRE_DIV_BYP   0x1
+#define RCC_CFGR_MCOPRE_DIV_2     0x2
+#define RCC_CFGR_MCOPRE_DIV_3     0x3
+#define RCC_CFGR_MCOPRE_DIV_4     0x4
+/* Note: MCOPRE_DIV can go from 1 - 15  */
+
+/* MCO1: Microcontroller clock output 1 */
+#define RCC_CFGR_MCO1_MASK        0x7
+#define RCC_CFGR_MCO1_SHIFT       22
+#define RCC_CFGR_MCO1_HSI         0x0
+#define RCC_CFGR_MCO1_LSE         0x1
+#define RCC_CFGR_MCO1_HSE         0x2
+#define RCC_CFGR_MCO1_PLL1        0x3
+#define RCC_CFGR_MCO1_HSI48       0x4
+#define RCC_CFGR_MCO_SHIFT        RCC_CFGR_MCO1_SHIFT
+#define RCC_CFGR_MCO_MASK         RCC_CFGR_MCO1_MASK
+
+/* RTCPRE: HSE division factor for RTC clock */
+#define RCC_CFGR_RTCPRE_SHIFT     8
+#define RCC_CFGR_RTCPRE_MASK      0x3f
+
+/* SWS: System clock switch status */
+#define RCC_CFGR_SWS_SHIFT        3
+#define RCC_CFGR_SWS_MASK         0x7
+#define RCC_CFGR_SWS_HSI          0x0
+#define RCC_CFGR_SWS_CSI          0x1
+#define RCC_CFGR_SWS_HSE          0x2
+#define RCC_CFGR_SWS_PLL1         0x3
+
+/* SW: System clock switch */
+#define RCC_CFGR_SW_SHIFT         0
+#define RCC_CFGR_SW_MASK          0x7
+#define RCC_CFGR_SW_HSI           0x0
+#define RCC_CFGR_SW_CSI           0x1
+#define RCC_CFGR_SW_HSE           0x2
+#define RCC_CFGR_SW_PLL1          0x3
+
+/**@}*/
+
+
+/* --- RCC_BDCR values ----------------------------------------------------- */
+#define RCC_BDCR_BDRST            (1 << 16)
+#define RCC_BDCR_RTCEN            (1 << 15)
+#define RCC_BDCR_RTCSEL_MASK      0x3
+#define RCC_BDCR_RTCSEL_SHIFT     8
+#define RCC_BDCR_RTCSEL_NONE      0
+#define RCC_BDCR_RTCSEL_LSE       1
+#define RCC_BDCR_RTCSEL_LSI       2
+#define RCC_BDCR_RTCSEL_HSE       3
+#define RCC_BDCR_LSEDRV_MASK      0x3
+#define RCC_BDCR_LSEDRV_SHIFT     3
+#define RCC_BDCR_LSEDRV_LOW       0
+#define RCC_BDCR_LSEDRV_MEDH      1 /* good job st */
+#define RCC_BDCR_LSEDRV_MEDL      2
+#define RCC_BDCR_LSEDRV_HIGH      3
+#define RCC_BDCR_LSEBYP           (1 << 2)
+#define RCC_BDCR_LSERDY           (1 << 1)
+#define RCC_BDCR_LSEON            (1 << 0)
+
+/* --- RCC_CSR values ------------------------------------------------------ */
+#define RCC_CSR_LSIRDY            (1 << 1)
+#define RCC_CSR_LSION             (1 << 0)
+
+extern uint32_t rcc_ahb_frequency;
+extern uint32_t rcc_apb1_frequency;
+extern uint32_t rcc_apb2_frequency;
+
+enum rcc_clock_3v3 {
+  RCC_CLOCK_3V3_216MHZ,
+  RCC_CLOCK_3V3_168MHZ,
+  RCC_CLOCK_3V3_120MHZ,
+  RCC_CLOCK_3V3_72MHZ,
+  RCC_CLOCK_3V3_48MHZ,
+  RCC_CLOCK_3V3_24MHZ,
+  RCC_CLOCK_3V3_END
+};
+
+// TODO: Customize this configuration based on the PLL options.
```

Comment:
no // comments and no todos please. what's going on here?

Reply:
These are meant as placeholders to fill in as there will need to be a structure like this in the file. It was not my intent to erase everything, and then add it back in. If TODO's are frowned upon, I'll just strip things and add back in.